### ISSUE TITLE
Feature: Unified Sentiment Processing Pipeline with Confidence Estimation and Usability-Oriented Mapping -GSOC-2026

### ISSUE THREAD

**Issue by HITESH-S-P** [ruxailab/RUXAILAB#1988]
<html>
<body>
<html><head></head><body><h1>Unified Sentiment Processing Pipeline with Confidence Estimation and Usability-Oriented Mapping - (Phase-1)</h1>
<p><strong>Labels:</strong> <code>enhancement</code> <code>gsoc-2026</code> <code>sentiment analysis</code>
<hr>
<h2>What needs to be built</h2>

<h3>1. Standard output schema</h3>
<p>
All modules (text, audio, facial) should return results in a common format. Each output must include:
</p>
<ul>
  <li><code>emotion</code> - predicted label</li>
  <li><code>confidence</code> - score between 0 and 1</li>
  <li><code>usability_label</code> - mapped UX-focused category (e.g. frustration, satisfaction)</li>
  <li><code>modality</code> - source (<code>text</code>, <code>audio</code>, <code>face</code>, <code>fused</code>)</li>
  <li><code>raw_scores</code> - optional probabilities for all classes</li>
</ul>

<h3>2. Confidence extraction</h3>
<p>
Model outputs (logits) should be converted into probabilities using <strong>softmax</strong>.
The confidence score must always be the probability of the top prediction and not raw values.
</p>
<p>
This will be important later when combining multiple modalities.
</p>

<h3>3. Emotion to usability mapping</h3>
<p>
We need a simple mapping from generic emotion labels to usability-focused categories.
</p>
<ul>
  <li>Clear and readable (dictionary-based)</li>
  <li>Easy to extend</li>
  <li>Separate from core logic</li>
</ul>

<table>
  <thead>
    <tr>
      <th>Emotion</th>
      <th>Usability</th>
    </tr>
  </thead>
  <tbody>
    <tr><td>anger, disgust, sadness</td><td>frustration</td></tr>
    <tr><td>fear</td><td>cognitive_load</td></tr>
    <tr><td>joy</td><td>satisfaction</td></tr>
    <tr><td>surprise</td><td>engagement</td></tr>
    <tr><td>neutral</td><td>neutral</td></tr>
    <tr><td>confusion</td><td>confusion</td></tr>
  </tbody>
</table>

<hr>
<h2>Expected output format</h2>
<pre><code class="language-json">{
  "emotion": "confusion",
  "confidence": 0.81,
  "usability_label": "cognitive_load",
  "modality": "text",
  "raw_scores": {
    "confusion": 0.81,
    "neutral": 0.12,
    "anger": 0.04,
    "joy": 0.03
  }
}
</code></pre>
<hr></html>
</body>
</html>
